src/game/stats.c

```diff
@@ -0,0 +1,198 @@
+#include "game/stats.h"
+
+#include "game/shell.h"
+#include "global/vars.h"
+#include "log.h"
+
+#include <stdio.h>
+#include <string.h>
+
+#define PIERRE_ITEMS 3
+#define SKATEKID_ITEMS 1
+#define COWBOY_ITEMS 1
+#define BALDY_ITEMS 1
+
+static int32_t m_LevelPickups = 0;
+static int32_t m_LevelKillables = 0;
+static int32_t m_LevelSecrets = 0;
+static bool m_KillableItems[MAX_ITEMS] = { 0 };
+static bool m_IfKillable[O_NUMBER_OF] = { 0 };
+
+int16_t m_PickupObjs[] = { O_PICKUP_ITEM1,   O_PICKUP_ITEM2,  O_KEY_ITEM1,
+                           O_KEY_ITEM2,      O_KEY_ITEM3,     O_KEY_ITEM4,
+                           O_PUZZLE_ITEM1,   O_PUZZLE_ITEM2,  O_PUZZLE_ITEM3,
+                           O_PUZZLE_ITEM4,   O_GUN_ITEM,      O_SHOTGUN_ITEM,
+                           O_MAGNUM_ITEM,    O_UZI_ITEM,      O_GUN_AMMO_ITEM,
+                           O_SG_AMMO_ITEM,   O_MAG_AMMO_ITEM, O_UZI_AMMO_ITEM,
+                           O_EXPLOSIVE_ITEM, O_MEDI_ITEM,     O_BIGMEDI_ITEM,
+                           O_SCION_ITEM,     O_SCION_ITEM2,   O_LEADBAR_ITEM,
+                           NO_ITEM };
+
+// Pierre has special trigger check
+int16_t m_KillableObjs[] = {
+    O_WOLF,     O_BEAR,        O_BAT,      O_CROCODILE, O_ALLIGATOR,
+    O_LION,     O_LIONESS,     O_PUMA,     O_APE,       O_RAT,
+    O_VOLE,     O_DINOSAUR,    O_RAPTOR,   O_WARRIOR1,  O_WARRIOR2,
+    O_WARRIOR3, O_CENTAUR,     O_MUMMY,    O_ABORTION,  O_DINO_WARRIOR,
+    O_FISH,     O_LARSON,      O_SKATEKID, O_COWBOY,    O_BALDY,
+    O_NATLA,    O_SCION_ITEM3, O_STATUE,   O_PODS,      O_BIG_POD,
+    NO_ITEM
+};
+
+static void Stats_CheckTriggers(FLOOR_INFO **floor_array);
+
+static void Stats_CheckTriggers(FLOOR_INFO **floor_array)
```

Comment:
Looks pretty good but I don't quite don't understand why do we need to create a copy of the floor data for the stats calc?

Reply:
Because of the way some doors work. There was a bug where the trigger check was missing triggers in a rare instance. See the screenshot below for a trigger that is part of the bigger room but is behind a closed door. When the doors are initialized, closed doors set the floor index to 0, so this trigger was never getting checked.

![image](https://user-images.githubusercontent.com/81546780/152653559-914ab578-3a6c-44d1-ad3e-581631636c20.png)